### DISCUSSION TITLE
[API/APP] Make user and role selectable in the filter rules

### DISCUSSION THREAD

**Discussion by Attacler** [directus/directus#10732] [Feature Requests]
Hi there,

So it might be a nice thing to be able to select the current user/role in filter rules.
In this case i could use this to hide fields based on the role of the current user.

**Comment by benhaynes** [comment]
Are you talking about [Dynamic Variables](https://docs.directus.io/configuration/filter-rules/#dynamic-variables)? Or how they should be entered into the Filter Component?

**Comment by Attacler** [comment]
No I'm more talking about using it inside the filter rules.
So you can say: current role must equal 123

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Reply by benhaynes** [reply]
Wait, can you not? @rijkvanzanten 🔔

**Reply by rijkvanzanten** [reply]
Dynamic Variables work the other way, it's `role _eq $CURRENT_ROLE`, not `$CURRENT_ROLE _eq 123` 🔁

**Reply by Attacler** [reply]
> Dynamic Variables work the other way, it's `role _eq $CURRENT_ROLE`, not `$CURRENT_ROLE _eq 123` 🔁

True!
So basicly if you want to show a field based on a the current role/user (or any variable on the user) it becomes quite difficult.

**Reply by Pachat** [reply]
@Attacler If you just want to show/hide a field based on the user's role, you just have to 
1. go to to Roles & Permissoins, 
2. chose the role
3. chose the collection
4. Use Custom 
5. and set the fields to be shown or not

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
